source/tutorials/nixos/nixos-configuration-on-vm.md

```diff
@@ -20,14 +20,37 @@ Virtual machines are a practical tool for debugging NixOS configurations.
 - A working [Nix installation](https://nixos.org/manual/nix/stable/installation/installation.html) on Linux, or [NixOS](https://nixos.org/manual/nixos/stable/index.html#sec-installation), with a graphical environment
 - Basic knowledge of the [Nix language](reading-nix-language)
 
-## Starting from the default NixOS configuration
+## Starting from a default NixOS configuration
 
-In this tutorial you will use the default configuration that is shipped with NixOS.[^nixosconf]
-[^nixosconf]: This [configuration template](https://github.com/NixOS/nixpkgs/blob/4e0525a8cdb370d31c1e1ba2641ad2a91fded57d/nixos/modules/installer/tools/tools.nix#L122-L226) is used.
+In this tutorial you will use a default configuration that is shipped with NixOS.
 
 :::{admonition} NixOS
 
 On NixOS, use the `nixos-generate-config` command to create a configuration file that contains some useful defaults and configuration suggestions.
+
+Beware that the result of this command depends on your current NixOS configuration.
+We can configure the generation of the 'nixos-generate-config' in a repoducable way in a nix-shell.
+We provide a configuration that is replicating nixos gnome graphical iso image.
```

Comment:
Hm, apparently that suggestion got lost in a merge conflict. Sorry for the second round:

```suggestion
The output of 'nixos-generate-config' can be made reproducible in a `nix-shell` environment. 
Here we provide a configuration that corresponds to the NixOS GNOME graphical ISO image:
```